### PR TITLE
CSM 123 : Create History REST Resource

### PR DESCRIPTION
**Pull request by maany**

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package org.openmrs.module.chartsearch.web.controller;
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java]: licence text need to be added

File: omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    @Override
+    public ChartSearchHistory getByUniqueId(String uuid) {
+        return Context.getService(ChartSearchService.class).getSearchHistoryByUuid(uuid);
+    }
+
+    @Override
+    protected void delete(ChartSearchHistory chartSearchHistory, String s, RequestContext requestContext) throws ResponseException {
+
+    }
+
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java]: Looks like you forgot to implement purge method

Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java]: Ooops! Just seen it below, ignore my comment

File: omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+        if (representation instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("searchPhrase");
+            description.addProperty("defaultSearch");
+            description.addProperty("patient", Representation.DEFAULT);
+            description.addProperty("historyOwner", Representation.DEFAULT);
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java]: patient and historyOwner should be REF representations

Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/chartsearch/web/controller/ChartSearchHistoryResource.java]: Did you see the above comment Maany?

**Comment by dkayiwa** [comment]
@maany do you think you will get time for this after releasing the platform 2.0 alpha?

**Comment by maany** [comment]
@dkayiwa yup! I'll make sure we close this once release is complete :)

**Comment by dkayiwa** [comment]
@maany now that the release has over delayed, do you think this should wait on it? :)

**Comment by maany** [comment]
@dkayiwa I will not be able to work on it in the next 3 weeks due to end-term exams. I'm closing the PR for now. I will make a new one after my exams :)

**Pull request closed without merging**